### DISCUSSION TITLE
EBSD Fill function changing Euler Angles of indexed points

### DISCUSSION THREAD

**Discussion by elegantEmily** [mtex-toolbox/mtex#935] [Ask Anything]
I am currently using the command "ebsd_Hraw = fill(ebsdG, grains)" on my EBSD data to fill in some unindexed points. However, when checking the Euler angles between my raw data and the filled data it appears MTEX is also changing the Euler Angles of indexed points as well.

Phi1 is the Euler angle that sees the most changes. Out of 154391 shared xy positions between the raw and filled data 149972 points are changed. I am using MTEX 5.6.0.

Visually the IPF maps look similar, but I expected the same xy points between the raw and filled EBSD data to have  matching Euler angles. Can someone explain the discrepancy?

I have attached a picture of some example points for phi1, 1st column is filled data, 2nd column is raw data, 3 column is the difference between the two.

![phi1NoMatch](https://user-images.githubusercontent.com/60235316/113758217-34a17c00-96d9-11eb-962a-b892454a0969.PNG)

Before Fill:
![RawData](https://user-images.githubusercontent.com/60235316/113758491-8ea24180-96d9-11eb-8b56-036ca7ac96a4.PNG)

After Fill:
![filledData](https://user-images.githubusercontent.com/60235316/113758500-9366f580-96d9-11eb-92fc-7797d8f2d751.PNG)

Thanks!
Emily

**Comment by kilir** [comment]
Hi,
have a look at the properties, you find an entry `oldId` and you can use this to id to select your initial data or the otehr way around:
```
mtexdata forsterite

[grains, ebsd.grainId]=ebsd.calcGrains;
ebsdf = ebsd.fill('grains')

ebsdf('id',115)
ebsd('id',ebsdf('id',115).prop.oldId)


ebsd('id',227)
ebsdf(ebsdf.prop.oldId == 227)
```
Hope this helps.

Cheers,
Rüdiger

**Comment by ralfHielscher** [comment]
The fill command should not alter the orientations. Are you sure the orientations have changed? Different Euler angles can still define the same orientation. 

Ralf.
